src/wallet/test/util.h

```diff
@@ -98,8 +98,9 @@ class MockableDatabase : public WalletDatabase
 public:
     MockableData m_records;
     bool m_pass{true};
+    bool m_read_only{false};
 
-    MockableDatabase(MockableData records = {}) : WalletDatabase(), m_records(records) {}
+    MockableDatabase(MockableData records = {}, bool read_only = false) : WalletDatabase(), m_records(records), m_read_only(read_only) {}
```

Comment:
In 6189a20491f71e3a6f499d2a510b38bb914e00d9 "test: Add tests for readonly database access"

`MockableBatch` should have `m_read_only` as well and it should be enforcing that in its `Write()`.